score.go

```diff
@@ -252,27 +274,24 @@ func scoreCheck(check check) {
 	}
 }
 
-func checkFails(check *check) bool {
+func checkOr(check *check, hint *hintItem) bool {
```

Comment:
Can we have a line or two to explain what `checkOr` and `checkAnd` are designed to do? it is a slightly less direct name than previously

Reply:
I saw that the functions were identical for checkFails/PassOverride because they were both OR/|| behavior, in that on the first true, return. Whereas pass is AND/&& because everything needs to be true otherwise return false. Sure I'll add a comment

Reply:
Actually good catch I messed that up and forgot to change it to take a list of conds rather than a check